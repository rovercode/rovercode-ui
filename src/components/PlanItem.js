import React from 'react';
import { Box, Typography } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import { hot } from 'react-hot-loader';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';

const Banner = withStyles((theme) => ({
  root: {
    backgroundColor: theme.palette.secondary.main,
    color: theme.palette.getContrastText(theme.palette.secondary.main),
    borderRadius: '4px 4px 0px 0px',
    maxWidth: '90%',
    marginLeft: 'auto',
    marginRight: 'auto',
  },
}))(Typography);

const PlanItem = ({ active, children, title }) => (
  <>
    {
    active === true ? (
      <Banner variant="subtitle2" align="center">
        <FormattedMessage
          id="app.plan_item.your_plan"
          description="Identifier for the subscription plan"
          defaultMessage="Your plan"
        />
      </Banner>
    ) : (null)
  }
    <Box p={2} border={1} borderRadius="borderRadius" boxShadow={2} borderColor="grey.300">
      <Typography variant="h5" align="center">
        {title}
      </Typography>
      {children}
    </Box>
  </>
);

PlanItem.defaultProps = {
  active: false,
};

PlanItem.propTypes = {
  active: PropTypes.bool,
  children: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
};

export default hot(module)(PlanItem);
