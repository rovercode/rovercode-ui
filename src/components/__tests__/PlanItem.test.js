import React from 'react';

import PlanItem from '../PlanItem';

describe('The PlanItem component', () => {
  test('renders active on the page with no errors', () => {
    const wrapper = mountWithIntl(
      <PlanItem title="Test Title" active>
        <div />
      </PlanItem>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('renders inactive on the page with no errors', () => {
    const wrapper = mountWithIntl(
      <PlanItem title="Test Title">
        <div />
      </PlanItem>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
