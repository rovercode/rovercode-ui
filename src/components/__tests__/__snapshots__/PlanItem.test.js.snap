// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The PlanItem component renders active on the page with no errors 1`] = `
<PlanItem
  active={true}
  title="Test Title"
>
  <WithStyles(WithStyles(ForwardRef(Typography)))
    align="center"
    variant="subtitle2"
  >
    <WithStyles(ForwardRef(Typography))
      align="center"
      classes={
        Object {
          "root": "WithStyles(ForwardRef(Typography))-root-1",
        }
      }
      variant="subtitle2"
    >
      <ForwardRef(Typography)
        align="center"
        classes={
          Object {
            "alignCenter": "MuiTypography-alignCenter",
            "alignJustify": "MuiTypography-alignJustify",
            "alignLeft": "MuiTypography-alignLeft",
            "alignRight": "MuiTypography-alignRight",
            "body1": "MuiTypography-body1",
            "body2": "MuiTypography-body2",
            "button": "MuiTypography-button",
            "caption": "MuiTypography-caption",
            "colorError": "MuiTypography-colorError",
            "colorInherit": "MuiTypography-colorInherit",
            "colorPrimary": "MuiTypography-colorPrimary",
            "colorSecondary": "MuiTypography-colorSecondary",
            "colorTextPrimary": "MuiTypography-colorTextPrimary",
            "colorTextSecondary": "MuiTypography-colorTextSecondary",
            "displayBlock": "MuiTypography-displayBlock",
            "displayInline": "MuiTypography-displayInline",
            "gutterBottom": "MuiTypography-gutterBottom",
            "h1": "MuiTypography-h1",
            "h2": "MuiTypography-h2",
            "h3": "MuiTypography-h3",
            "h4": "MuiTypography-h4",
            "h5": "MuiTypography-h5",
            "h6": "MuiTypography-h6",
            "noWrap": "MuiTypography-noWrap",
            "overline": "MuiTypography-overline",
            "paragraph": "MuiTypography-paragraph",
            "root": "MuiTypography-root WithStyles(ForwardRef(Typography))-root-1",
            "srOnly": "MuiTypography-srOnly",
            "subtitle1": "MuiTypography-subtitle1",
            "subtitle2": "MuiTypography-subtitle2",
          }
        }
        variant="subtitle2"
      >
        <h6
          className="MuiTypography-root WithStyles(ForwardRef(Typography))-root-1 MuiTypography-subtitle2 MuiTypography-alignCenter"
        >
          <FormattedMessage
            defaultMessage="Your plan"
            id="app.plan_item.your_plan"
          >
            Your plan
          </FormattedMessage>
        </h6>
      </ForwardRef(Typography)>
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(WithStyles(ForwardRef(Typography)))>
  <Styled(MuiBox)
    border={1}
    borderColor="grey.300"
    borderRadius="borderRadius"
    boxShadow={2}
    p={2}
  >
    <div
      className="MuiBox-root MuiBox-root-2"
    >
      <WithStyles(ForwardRef(Typography))
        align="center"
        variant="h5"
      >
        <ForwardRef(Typography)
          align="center"
          classes={
            Object {
              "alignCenter": "MuiTypography-alignCenter",
              "alignJustify": "MuiTypography-alignJustify",
              "alignLeft": "MuiTypography-alignLeft",
              "alignRight": "MuiTypography-alignRight",
              "body1": "MuiTypography-body1",
              "body2": "MuiTypography-body2",
              "button": "MuiTypography-button",
              "caption": "MuiTypography-caption",
              "colorError": "MuiTypography-colorError",
              "colorInherit": "MuiTypography-colorInherit",
              "colorPrimary": "MuiTypography-colorPrimary",
              "colorSecondary": "MuiTypography-colorSecondary",
              "colorTextPrimary": "MuiTypography-colorTextPrimary",
              "colorTextSecondary": "MuiTypography-colorTextSecondary",
              "displayBlock": "MuiTypography-displayBlock",
              "displayInline": "MuiTypography-displayInline",
              "gutterBottom": "MuiTypography-gutterBottom",
              "h1": "MuiTypography-h1",
              "h2": "MuiTypography-h2",
              "h3": "MuiTypography-h3",
              "h4": "MuiTypography-h4",
              "h5": "MuiTypography-h5",
              "h6": "MuiTypography-h6",
              "noWrap": "MuiTypography-noWrap",
              "overline": "MuiTypography-overline",
              "paragraph": "MuiTypography-paragraph",
              "root": "MuiTypography-root",
              "srOnly": "MuiTypography-srOnly",
              "subtitle1": "MuiTypography-subtitle1",
              "subtitle2": "MuiTypography-subtitle2",
            }
          }
          variant="h5"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
          >
            Test Title
          </h5>
        </ForwardRef(Typography)>
      </WithStyles(ForwardRef(Typography))>
      <div />
    </div>
  </Styled(MuiBox)>
</PlanItem>
`;

exports[`The PlanItem component renders inactive on the page with no errors 1`] = `
<PlanItem
  active={false}
  title="Test Title"
>
  <Styled(MuiBox)
    border={1}
    borderColor="grey.300"
    borderRadius="borderRadius"
    boxShadow={2}
    p={2}
  >
    <div
      className="MuiBox-root MuiBox-root-3"
    >
      <WithStyles(ForwardRef(Typography))
        align="center"
        variant="h5"
      >
        <ForwardRef(Typography)
          align="center"
          classes={
            Object {
              "alignCenter": "MuiTypography-alignCenter",
              "alignJustify": "MuiTypography-alignJustify",
              "alignLeft": "MuiTypography-alignLeft",
              "alignRight": "MuiTypography-alignRight",
              "body1": "MuiTypography-body1",
              "body2": "MuiTypography-body2",
              "button": "MuiTypography-button",
              "caption": "MuiTypography-caption",
              "colorError": "MuiTypography-colorError",
              "colorInherit": "MuiTypography-colorInherit",
              "colorPrimary": "MuiTypography-colorPrimary",
              "colorSecondary": "MuiTypography-colorSecondary",
              "colorTextPrimary": "MuiTypography-colorTextPrimary",
              "colorTextSecondary": "MuiTypography-colorTextSecondary",
              "displayBlock": "MuiTypography-displayBlock",
              "displayInline": "MuiTypography-displayInline",
              "gutterBottom": "MuiTypography-gutterBottom",
              "h1": "MuiTypography-h1",
              "h2": "MuiTypography-h2",
              "h3": "MuiTypography-h3",
              "h4": "MuiTypography-h4",
              "h5": "MuiTypography-h5",
              "h6": "MuiTypography-h6",
              "noWrap": "MuiTypography-noWrap",
              "overline": "MuiTypography-overline",
              "paragraph": "MuiTypography-paragraph",
              "root": "MuiTypography-root",
              "srOnly": "MuiTypography-srOnly",
              "subtitle1": "MuiTypography-subtitle1",
              "subtitle2": "MuiTypography-subtitle2",
            }
          }
          variant="h5"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
          >
            Test Title
          </h5>
        </ForwardRef(Typography)>
      </WithStyles(ForwardRef(Typography))>
      <div />
    </div>
  </Styled(MuiBox)>
</PlanItem>
`;
